internal/constants/scripts.go

```diff
@@ -66,140 +70,149 @@ function map_host_macs_to_interfaces() {
-      then
-        # get host interface name
-        host_iface=${host_macs_to_hw_iface[$mac]}
-        if [ -z "$host_iface" ]
-        then
-          echo "Warning: Mapping file contains MAC Address '$mac' (for logical interface name '$nic') that doesn't exist on the host"
-          continue
-        fi
-
-        # replace logical interface name with host interface name
-        sed -i -e "s/=$nic$/=$host_iface/g" -e "s/=$nic\./=$host_iface\./g" $nmconn_file
-        echo "Info: Using logical interface name '$nic' for interface with Mac address '$mac', updated $nmconn_file)"
-      fi
+      mac=${line%%=*}
+      nic=${line#*=}
+      logical_mac_nic_map[${mac^^}]=$nic
     done
   done
 }
 
-function copy_nmconnection_files_to_nm_config_dir() {
-  echo "Info: Copying nmconnection files to $ETC_NETWORK_MANAGER"
-  for nmconn_file in $(ls -1 ${host_dir}/*.nmconnection)
-  do
-    # rename nmconnection files based on the actual interface name
-    filename=$(basename $nmconn_file)
-    prefix="${filename%%.*}"
-    extension="${filename#*.}"
-    if [ ! -z "${logical_nic_mac_map[$prefix]}" ]
-    then
-      dir_path=$(dirname $nmconn_file)
-      mac_address=${logical_nic_mac_map[$prefix]}
-      host_iface=${host_macs_to_hw_iface[$mac_address]}
-      if [ ! -z "$host_iface" ]
-      then
-        mv $nmconn_file "${dir_path}/${host_iface}.${extension}"
-        echo "Info: Copied $nmconn_file to $dir_path/$host_iface/$extension"
-        continue
-      fi
-      echo "Warning: Mapping for '$mac_address' was not present while attempting to copy $nmconn_file to $dir_path/$host_iface/$extension"
-    fi
-  done
-
-  mkdir -p "${ETC_NETWORK_MANAGER}"
-  cp ${host_dir}/*.nmconnection ${ETC_NETWORK_MANAGER}/
-  echo "Info: Copied all nmconnection files from $host_dir to $ETC_NETWORK_MANAGER"
-}
 
 echo "PreNetworkConfig Start"
+
+# Get the mac to host nic mapping from local machine
 map_host_macs_to_interfaces
-find_host_directory_by_mac_address
 
-// Make sure we do not run any of the following functions if there was no matching config.
-if [ "$host_dir" ]
-  then
+# Get mac to local nic mapping from all mac_mapping.ini files in host directories
+map_mac_logical_nic_mapping
+
+# Process all relevant files and put them in work_dir
+process_host_directories_by_mac_address
+
+# Make sure we do not run any of the following functions if there was no matching config.
+if [ "$(ls -A $work_dir )" ];  then
     # Remove default connection file create by NM(nm-initrd-generator). This is a WA until
     # NM is back to supporting priority between nmconnections.
     #
     # Note that when this is intended mostly for the full ISO scenario. In the minimal ISO scenario
     # this script runs before the 'coreos-coreos-copy-firstboot-network' service, and that script
     # already removes the connection files, but it doesn't hurt to do it again.
-    rm -f ${ETC_NETWORK_MANAGER}/*
+    mkdir -p ${ETC_NETWORK_MANAGER}
```

Comment:
Why did this need to be added?

Reply:
It was in different location

Reply:
Ah okay, it probably makes more sense to be right before the copy, right?

Reply:
Yes